### ISSUE TITLE
Unwanted query results

### ISSUE THREAD

**Issue by Cardenaz** [cngeru/georange#2]
There seems to be a problem with the geo query. No matter the distance in longitude, the query picks everything that's beyond the radius limit. It does work when applied laterally, meaning it discards results in latitude. I suggest to try to create a scenario where you have several items with similar latitude., but vastly different values in longitude. You will notice that the query picks up results that are double the required distance. 

/W

**Comment by cngeru** [comment]
Can you share some code ?

**Comment by Cardenaz** [comment]
Yes. It seemed to work perfectly, but when I tested this query further I noticed that it sometimes includes results that are more than 30km away(longitudian direction). The firestore documents consists of a position object, containing a geohash and a geopoint. The query is based on the geohash property of the position object. 
Inside our database class we have a geoStream function with the following query defined: 
`  final GeoRange georange = GeoRange(); Range range = georange.geohashRange(lat, lng, distance: 7);
    Query query = FirebaseFirestore.instance.collection(path).where("position.geohash", isGreaterThanOrEqualTo:range.lower).where("position.geohash", isLessThanOrEqualTo: range.upper).limit(10);
`
 The stream is later returned from: 
`final singleStream = StreamProvider.autoDispose<List<MockModel>>((ref) {
     final database = ref.watch(databaseProvider);
    final location = ref.watch(loc);
    double lat = location.position.latitude;
    double lng = location.position.longitude;
    return database.streamGeoStream(
        lat: lat,
        lng: lng,
  );
  });`

Again, it works in a lot of cases but we noticed that unwanted results were slipping in despite their large distance. 
/W
